src/components/stateless/specific/modals/ConfirmBuyModal/ConfirmBuyModal.jsx

```diff
@@ -7,15 +7,26 @@ import Label from "../../../generic/Label/Label";
 import Button from "../../../generic/Button/Button";
 import AttributeValueFieldContainer from "../../containers/AttributeValueFieldContainer/AttributeValueFieldContainer";
 import {showModal, hideModal} from '../../../../../redux/actions/ModalAction';
+import  { escrow } from "../../../../../redux/actions/PurchaseAction";
+
 
 const mapStateToProps = (state) => {
     return {
+        offer: state.offer,
+        quantity: state.modal.args.quantity,
+        token: state.token.token?state.token.token.contract:null,
```

Comment:
Let's move token creation to when the contract is created (for now just always create a new token, in the future there should be a form that allows either to create new token, or give address of existing token)